### DISCUSSION TITLE
Can sp-rest-proxy be run without it using a private.json file?

### DISCUSSION THREAD

**Discussion by davea38** [koltyakov/sp-rest-proxy#170] [General]
I want to feed sp-rest-proxy the info held within private.json via environment variables on startup. 

I require there to be no user input as the proxy will be running with a service account in a docker container.

My private.json usually looks like this:

```json
{
  "siteUrl": "https://quantum.computacenter.com",
  "strategy": "OnPremiseUserCredentials",
  "username": "foo@bar.com",
  "password": "<password>"
}
```

Do I need to override authConfigSettings / authOptions to do this??

**Comment by koltyakov** [comment]
Hi @davea38, 

Yes, this is possible. Please check [the way providing node-sp-auth settigns as env variables](https://github.com/koltyakov/node-sp-auth-config#environment-variables).

**Reply by davea38** [reply]
Thank you @koltyakov .

It looks like I need to set ```SPAUTH_SITEURL```, ```SPAUTH_STRATEGY```, ```SPAUTH_USERNAME``` and ```SPAUTH_PASSWORD``` and set headless mode to true.

Your test file has an example of this https://github.com/koltyakov/node-sp-auth-config/blob/f5c40e1158595c9968c6afecb6c0640213c2bf5a/test/env.ts#L7
